src/Corporation/Actions.ts

```diff
@@ -198,125 +198,120 @@ export function acceptInvestmentOffer(corporation: Corporation): void {
   corporation.investorShares += investShares;
 }
 
-export function sellMaterial(material: Material, amount: string, price: string): void {
-  if (price === "") price = "0";
-  if (amount === "") amount = "0";
-  let cost = price.replace(/\s+/g, "");
-  cost = cost.replace(/[^-()\d/*+.MPe]/g, ""); //Sanitize cost
-  let temp = cost.replace(/MP/, "1.234e5");
+export function convertPriceString(price: string): string | number {
+  // Replace whitespace characters.
+  let sanitizedPrice = price.replace(/\s+/g, "");
+  /**
+   * Replace invalid characters. Only accepts:
+   * - Digit characters
+   * - 4 most basic algebraic operations (+ - * /)
+   * - Parentheses
+   * - Dot character
+   * - e
+   * - MP
+   */
+  sanitizedPrice = sanitizedPrice.replace(/[^\d+\-*/().eMP]/g, "");
+
+  // // If the input does not contain "MP", we only need to convert it to a number.
+  if (!sanitizedPrice.includes("MP")) {
+    const priceAsNumber = parseFloat(sanitizedPrice);
+    if (!Number.isFinite(priceAsNumber)) {
+      throw new Error(`Invalid value for sell price field: ${price}`);
+    }
+    return priceAsNumber;
+  }
+
+  // Replace MP with an arbitrary number.
+  const temp: string = sanitizedPrice.replace(/MP/g, "1.234e5");
+  let evaluatedTemp: unknown;
   try {
-    if (temp.includes("MP")) throw "Only one reference to MP is allowed in sell price.";
-    temp = eval?.(temp);
+    evaluatedTemp = eval?.(temp);
+    if (typeof evaluatedTemp !== "number" || !Number.isFinite(evaluatedTemp)) {
+      throw new Error(`Evaluated value is not a valid number: ${evaluatedTemp}`);
+    }
   } catch (error) {
-    throw new Error("Invalid value or expression for sell price field", { cause: error });
+    throw new Error(`Invalid value or expression for sell price field: ${error}`, { cause: error });
+  }
+  // Use sanitized price.
+  return sanitizedPrice;
+}
+
+export function convertAmountString(amount: string, max: number, prod: number, inv: number) {
+  // Replace whitespace characters.
+  let sanitizedAmount = amount.replace(/\s+/g, "");
+  /**
+   * Replace invalid characters. Only accepts:
+   * - Digit characters
+   * - 4 most basic algebraic operations (+ - * /)
+   * - Parentheses
+   * - Dot character
+   * - e
+   * - MAXPRODINV
+   */
+  sanitizedAmount = sanitizedAmount.replace(/[^\d+\-*/().eMAXPRODINV]/g, "");
+
+  // // If the input does not contain "MAX", "PROD", and "INV", we only need to convert it to a number.
+  if (!sanitizedAmount.includes("MAX") && !sanitizedAmount.includes("PROD") && !sanitizedAmount.includes("INV")) {
+    const amountAsNumber = parseFloat(sanitizedAmount);
+    if (!Number.isFinite(amountAsNumber)) {
+      throw new Error(`Invalid value for sell quantity field: ${amount}`);
+    }
+    return amountAsNumber;
   }
 
-  if (temp == null || isNaN(parseFloat(temp))) {
-    throw new Error("Invalid value or expression for sell price field");
+  let temp = sanitizedAmount.replace(/MAX/g, max.toString());
```

Comment:
Can't these be placeholders to avoid needing to pass in real values?

Also, the placeholder should be negative where that is a possible value, to catch parsing issues caused by unary minus.